docs/docs/integrating-with-other-libraries.md

```diff
@@ -0,0 +1,270 @@
+
+```js
+class Chosen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleUpdate = this.handleUpdate.bind(this);
+  }
+
+  handleUpdate(event) {
+    this.props.update(event.target.value);
+  }
+
+  componentDidMount() {
+    this.$el = $(this.el);
+    this.$el.chosen();
+    this.$el.on('change', this.handleUpdate);
+  }
+
+  componentDidUpdate() {
+    this.$el.trigger('chosen:updated');
+  }
+
+  componentWillUnmount() {
+    this.$el.off('change', this.handleUpdate);
+    this.$el.chosen('destroy');
+  }
+
+  render() {
+    return (
+      <select ref={el => this.el = el}>
+        {this.props.children}
+      </select>
+    );
+  }
+}
+```
+
+1. Change listeners need to be setup inside `componentDidMount()` and torn down in `componentWillUnmount` as the events are emitted by the plugin use jQuery, which is outside the React event system.
+
+2. Chosen needs to be notified of any changes to the original select or its children. This is done in `componentWillUpdate()` by triggering a `'chosen:updated'` event.
+
+3. When the component is unmounted, use the cleanup method provided by Chosen. This removes the custom select control and restores the actual `<select>` as well as removing any event listeners the plugin registered.
+
+[Try it on CodePen.](http://codepen.io/wacii/pen/ygzxjG?editors=0010)
+
+## Integrating with Other View Libraries
+
+React can be easily embedded into other applications thanks to the flexibility of `ReactDOM.render()`. While often used once at startup to load a single React application into the DOM, `ReactDOM.render()` can be called multiple times, both to create multiple React applications and to update existing ones. This allows applications to be rewritten in React piece by piece.
+
+### Replacing String Based Rendering with React
+
+A common pattern in older web applications is to describe chunks of the DOM as a string and insert it into the DOM like so: `$el.html(htmlString)`. These points in a codebase are perfect for introducing React. Just rewrite the string based rendering as a React component.
+
+So the following jQuery implementation...
+```js
+const htmlString = '<button id="jquery-btn">jQuery</button>';
+$el.html(htmlString);
+$('jquery-btn').click(() => window.alert('jQuery button'));
+```
+
+...could be rewritten using a React Component.
+```js
+function Button() {
+  return <button id='react-btn'>React</button>;
+}
+ReactDOM.render(
+  <Button />,
+  document.getElementById('react-container'),
+  () => {
+    $('#react-btn').click(() => window.alert('React button'));
+  }
+);
+```
+
+From here you could start moving more logic into the component and begin adopting more common React practices.
```

Comment:
Maybe explicitly say "For example, in components it is best not to rely on IDs because the same component can be rendered multiple times. Instead, we will use React event system:" and link that to events doc.